spec/support/fake_yarn_matchers.rb

```diff
@@ -0,0 +1,25 @@
+# FakeYarn records yarn information through a fake "bin/yarn"
+# binary. This RSpec matcher relies on that functionality.
+RSpec::Matchers.define :have_yarned do |actual_yarn_args = ""|
+  match do |file_str|
+    raise 'Use expect("package.json")' if file_str != "package.json"
+
+    fake_yarn = FakeYarn.new
+
+    @yarn_ran = fake_yarn.ran?
+    @given_yarn_args = fake_yarn.arguments
+
+    @yarn_ran && @given_yarn_args == actual_yarn_args
+  end
+
+  failure_message do
+    if !@yarn_ran
+      "The generator did not call yarn!"
```

Comment:
Do you mind dropping the `!` from these strings? Pedantic, maybe, but I don't like shouting when I'm trying to debug a problem.

Reply:
Not at all! I feel you.